packages/realm/src/app-services/User.ts

```diff
@@ -256,7 +284,21 @@ export class User<
    *                       .find({color: 'blue'});
    */
   mongoClient(serviceName: string): unknown {
-    throw new Error("Not yet implemented");
+    return {
+      get serviceName() {
+        return serviceName;
+      },
+      db: (dbName: string) => {
+        return {
+          get name() {
+            return dbName;
+          },
+          collection: (collectionName: string) => {
+            return new MongoClient(this as User, serviceName, dbName, collectionName);
```

Comment:
Ditto question about cast. If it is warning without the cast, I'd be a bit concerned that `this` is bound to something else here. Can you check what type TS thinks `this` is without the cast?

Reply:
<img width="452" alt="image" src="https://user-images.githubusercontent.com/44238293/201337408-8162fe91-f6a3-48d0-92d4-ef18938643e3.png">
this is the error generated from removing the case

Reply:
A similar error is caused by the `return createFactory(this as User, undefined)`